### PR TITLE
feat(ModelHelpers): add component mesh queries on edges

### PR DESCRIPTION
**Pull request by BotellaA**

### CODE REVIEW

File: src/geode/model/helpers/component_mesh_vertices.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        if( unique_vertices0.empty() || unique_vertices1.empty() )
+        {
+            return {};
+        }
+        std::vector< MeshComponentVertexIntersection > result;
+        result.reserve(
+            std::min( unique_vertices0.size(), unique_vertices1.size() ) );
+        for( const auto& mcv0 : unique_vertices0 )
+        {
+            for( const auto& mcv1 : unique_vertices1 )
```
Comment (panquez) [inline src/geode/model/helpers/component_mesh_vertices.cpp:44]: double for loop, or for + absl::c_find ?

File: src/geode/model/helpers/component_mesh_vertices.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            {
+                if( mcv0.component_id == mcv1.component_id
+                    && mcv0.vertex != mcv1.vertex )
+                {
+                    result.emplace_back(
+                        mcv0.component_id, mcv0.vertex, mcv1.vertex );
```
Comment (panquez) [inline src/geode/model/helpers/component_mesh_vertices.cpp:50]: if double for loop, break here

File: include/geode/model/helpers/component_mesh_vertices.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+              vertex1{ vertex1_in }
+        {
+        }
+
+        ComponentID component_id;
+        index_t vertex0;
+        index_t vertex1;
+    };
+
+    std::vector< MeshComponentVertexIntersection >
```
Comment (panquez) [inline include/geode/model/helpers/component_mesh_vertices.h:54]: when I read the name of the function, I thought that it returns only the MCV that are in both Spans. That does not. What does it do?

File: src/geode/model/helpers/component_mesh_vertices.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ */
+
+#include <geode/model/helpers/component_mesh_vertices.h>
+
+#include <geode/model/mixin/core/vertex_identifier.h>
+
+namespace geode
+{
+    std::vector< MeshComponentVertexIntersection >
```
Comment (panquez) [inline src/geode/model/helpers/component_mesh_vertices.cpp:30]: My comments on this file were for computing the intersection of the two spans as explained in the header file

File: src/geode/model/helpers/component_mesh_vertices.cpp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        }
+        std::vector< MeshComponentVertexIntersection > result;
+        result.reserve(
+            std::min( unique_vertices0.size(), unique_vertices1.size() ) );
+        for( const auto& mcv0 : unique_vertices0 )
+        {
+            for( const auto& mcv1 : unique_vertices1 )
+            {
+                if( mcv0.component_id == mcv1.component_id
+                    && mcv0.vertex != mcv1.vertex )
```
Comment (panquez) [inline src/geode/model/helpers/component_mesh_vertices.cpp:47]: != ?

File: tests/model/test-component-mesh-edges.cpp [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                geode::TriangulatedSurface2D::type_name_static() ) );
+        auto surface_builder =
+            builder.surface_mesh_builder< geode::TriangulatedSurface2D >(
+                surface_id );
+        surface_builder->create_point( points[0] );
+        surface_builder->create_point( points[1] );
+        surface_builder->create_point( points[2] );
+        surface_builder->create_point( points[4] );
+        surface_builder->create_point( points[5] );
+        surface_builder->create_point( points[3] );
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply tests/model/test-component-mesh-edges.cpp:67]: yes

File: include/geode/model/helpers/component_mesh_edges.h [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    };
+
+    struct SectionComponentMeshEdges : public ModelComponentMeshEdges
+    {
+    };
+
+    SectionComponentMeshEdges opengeode_model_api component_mesh_edges(
+        const Section& section, const Line2D& line, index_t edge );
+
+    SectionComponentMeshEdges opengeode_model_api component_mesh_edges(
```
[the remark this answers is not in the document]
Comment (BotellaA) [reply include/geode/model/helpers/component_mesh_edges.h:85]: I though about the same API than VertexIdentifier where the are overloads with the same names but different goals

File: tests/model/test-component-mesh-edges.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (panquez) [reply tests/model/test-component-mesh-edges.cpp:67]: I mean the Section is not valid. But is ok for the test

File: include/geode/model/helpers/component_mesh_vertices.h (hunk 3 above)
Comment (BotellaA) [reply include/geode/model/helpers/component_mesh_vertices.h:54]: map<ComponentID, vector< pair<index_t> >

Comment (BotellaA) [reply include/geode/model/helpers/component_mesh_vertices.h:54]: ComponentMeshVertexPairs component_mesh_vertex_pairs

**Review by panquez — APPROVED** [review]

**Pull request merged**